### DISCUSSION TITLE
How to process a json file using `docker-compose`?

### DISCUSSION THREAD

**Discussion by diego898** [ArchiveBox/ArchiveBox#1093] [General]
For context, I'm following along this guide: https://www.ecliptik.com/bookmarking-with-raindrop/

I have a few bookmarks, took raindrops exported .html file, and following the author's advice used the script to create this easy `.json` file from it. 

I'm now trying to import that JSON file into archive box. The file itself is very simple:

`data/sources/import-20230206.json`
```
[
{ "url": "https://news.ycombinator.com/item?id=34665738", "tags": "February 6 2023,first" },
{ "url": "https://news.ycombinator.com/item?id=34667067", "tags": "February 6 2023,first" },
{ "url": "https://news.ycombinator.com/item?id=34657535#34660922", "tags": "February 6 2023,first" },
{ "url": "https://news.ycombinator.com/item?id=23153778", "tags": "February 6 2023,first" }
]
```

When following along with the docs I try:

```
~/archivebox
❯ docker-compose run archivebox add < data/sources/import-20230206.json
the input device is not a TTY
```

and 

```
~/archivebox took 5s
❯ cat data/sources/import-20230206.json | docker-compose run archivebox add
the input device is not a TTY
```

Both don't work.

Then I try without the `<` or `cat`  and get:

```
~/archivebox took 3s
❯ docker-compose run archivebox add data/sources/import-20230206.json
[i] [2023-02-06 19:29:08] ArchiveBox v0.6.2: archivebox add data/sources/import-20230206.json
    > /data

[+] [2023-02-06 19:29:10] Adding 1 links to index (crawl depth=0)...
    > Saved verbatim input to sources/1675711750-import.txt
    > Parsed 0 URLs from input (Failed to parse)
    > Found 0 new URLs not already in index

[*] [2023-02-06 19:29:10] Writing 0 links to main index...
    √ ./index.sqlite3
```

which I can't really understand. 

Any help would be great!

**Comment by pirate** [comment]
You need `-T` to pass stdin to docker-compose commands, try this: `docker-compose run -T archivebox add < data/sources/import-20230206.json`.

https://dev.to/codewithcats/docker-compose-exec-using-stdin-as-an-input-46lh

**Reply by diego898** [reply]
Ah ok thank you!

**Comment by diego898** [comment]
Thank you! I think the documentation needs to be updated to reflect that! (Maybe a target for me!)

Also, this import work, but none of the tags in the JSON file made it through:

```
~/archivebox
❯ docker-compose run -T archivebox add < data/sources/import-20230206.json
[i] [2023-02-07 16:49:30] ArchiveBox v0.6.2: archivebox add
    > /data

[+] [2023-02-07 16:49:32] Adding 7 links to index (crawl depth=0)...
    > Saved verbatim input to sources/1675788572-import.txt
    > Parsed 8 URLs from input (Generic TXT)
    > Found 2 new URLs not already in index

[*] [2023-02-07 16:49:32] Writing 2 links to main index...
    √ ./index.sqlite3

[▶] [2023-02-07 16:49:32] Starting archiving of 2 snapshots in index...

[+] [2023-02-07 16:49:32] "news.ycombinator.com/item?id=34657535#34660922"
    https://news.ycombinator.com/item?id=34657535#34660922
    > ./archive/1675788572.642873
      > title
      > favicon
      > headers
      > singlefile
      > pdf
      > screenshot
      > dom
      > wget
      > readability
      > mercury
      > media
      > archive_org
        26 files (4.7 MB) in 0:01:36s

[+] [2023-02-07 16:51:09] "news.ycombinator.com/item?id=23153778"
    https://news.ycombinator.com/item?id=23153778
    > ./archive/1675788572.644349
      > title
      > favicon
      > headers
      > singlefile
      > pdf
      > screenshot
      > dom
      > wget
      > readability
      > mercury
      > media
      > archive_org
        26 files (2.0 MB) in 0:01:15s

[√] [2023-02-07 16:52:24] Update of 2 pages complete (2.87 min)
    - 0 links skipped
    - 2 links updated
    - 0 links had errors

    Hint: To manage your archive in a Web UI, run:
        archivebox server 0.0.0.0:8000
```

**Comment by diego898** [comment]
@pirate - just to clarify. Is it bug that none of the tags in the JSON file made it in? If so, I can file a new bug report.

**Reply by rmohns** [reply]
Hi, I’ve run into this as well. New installation of Archivebox (docker-compose). I’m working with exports from Pinboard, and tags are not imported from _any_ of the JSON , XML, or HTML export formats. Should I write up as a bug report? (I don’t want to create extra work if it’s already in progress… but my search of issues only turned up some years-old items that are marked fixed.)

**Reply by pirate** [reply]
Can you check that the issue persists on the latest 0.6.3 `dev` branch first: https://github.com/ArchiveBox/ArchiveBox#install-and-run-a-specific-github-branch

We had some tag fixes merged since the last (now embarrassingly old) 0.6.2 release. If you are still not seeing tags importing on 0.6.3 please open a new issue, thanks!
